DOC Remove random_state note from TimeSeriesSplit, and add where needed

will7142 opened the pull request:
TimeSeriesSplit does not use a randomized CV splitter and therefore should not include a note regarding "random state". This issue also occurs in the following classes: LeavePOut, LeaveOneOut, LeavePGroupsOut, LeaveOneGroupOut, and GroupKFold. Lastly, a note regarding randomized CV splitters should be added to: RepeatedStratifiedKFold and RepeatedKFold.

<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


#### What does this implement/fix? Explain your changes.


#### Any other comments?


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

will7142 committed:
Remove random_state note from TimeSeriesSplit

TimeSeriesSplit does not use a randomized CV splitter and therefore should not include a note regarding "random state". This issue also occurs in the following classes: LeavePOut, LeaveOneOut, LeavePGroupsOut, LeaveOneGroupOut, and GroupKFold. Lastly, a note regarding randomized CV splitters should be added to: RepeatedStratifiedKFold and RepeatedKFold.

jnothman reviewed (commented):
Thanks. Please put all those changes in one PR.

will7142 committed:
Fixes 9 random_state notes in model_selection

There are 14 model_selection Splitter Classes (plus train_test_split). The documentation in the "Notes" section for many of these classes regarding "random_state" was either erroneously included or omitted. This PR fixes this issue.
DETAILS:
Classes where random_state note was removed: GroupKFold, LeaveOneGroupOut, LeavePGroupsOut, LeaveOneOut, LeavePOut, Time SeriesSplit.
Classes where random_state note was added: RepeatedKFold, RepeatedStratifiedKFold, train_test_split
Classes where no change was made: GroupShuffleSplit, KFold, PredefinedSplit, ShuffleSplit, StratifiedKFold, StratifiedShuffleSplit

jnothman commented on sklearn/model_selection/_split.py:
```diff
@@ -1971,6 +1967,12 @@ def train_test_split(*arrays, **options):
+    Notes
+    -----
+    Randomized CV splitters may return different results for each call of
+    split. You can make the results identical by setting ``random_state``
```
Not applicable here: this is not a sitter with a `split` method

will7142 replied:
Good catch! Thank you. I will remove this update

jnothman reviewed (commented):
Otherwise LGTM

will7142 committed:
Fixes 8 random_state notes in model_selection

There are 14 model_selection Splitter Classes. The documentation in the "Notes" section for many of these classes regarding "random_state" was either erroneously included or omitted. This PR fixes this issue.
DETAILS:
Classes where random_state note was removed: GroupKFold, LeaveOneGroupOut, LeavePGroupsOut, LeaveOneOut, LeavePOut, Time SeriesSplit.
Classes where random_state note was added: RepeatedKFold, RepeatedStratifiedKFold
Classes where no change was made: GroupShuffleSplit, KFold, PredefinedSplit, ShuffleSplit, StratifiedKFold, StratifiedShuffleSplit

jnothman reviewed (approved):
LGTM, thanks!
